### ISSUE TITLE
[RFC] database_cleanup check filestore

### ISSUE THREAD

**Issue by legalsylvain** [OCA/server-tools#2309]
Hi all,

I have a V12 database (and associated filestore) that comes from V5.0

I'd like to check if the filestore is consistent. I mean :
- maybe some ir.attachment  are linked to a disappeared file in the filestore
- maybe some files exists but related ir.attachment does'nt.

Do you have similar needs ? 

I think that this feature could be part of the ``database_cleanup`` module.

CC : @StefanRijnhart, @guewen, @hbrunn 

thanks for your comments.

**Comment by hbrunn** [comment]
yes, I think that's a good one. Only issue I have is that it's hard to determine the right thing to do for the general case, do we want to delete ir.attachment records that don't have files or mark them for investigation? Probably you'll need some UI to ask the user what to do there, or make it very clear what it does ("Delete attachment records without file" vs "Cleanup attachments")

**Comment by legalsylvain** [comment]
Hi @hbrunn. Thanks a lot for your answer.
I'm not sure to understand your proposal.

in all database_cleanup features there are two steps  : 
1. one to display the errors
2. one to "fix" the errors

- In that case, users can stop at the end of the step 1, do an extract of the results and investigate. (the same for the other cleanup (columns table, extra tables, etc...) 
- for the fix, I don't see other kind of way to fix the problem except to delete the attachment (Do you have another idea ?)

thanks.

**Comment by hbrunn** [comment]
ah yes, I was thinking of the 'purge all' button, but indeed, you don't have to click that. Yes, makes sense
